statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,257 @@
+        elif vname is None:
+            vname = np.ones([self.n,1])
+        else:
+            vname = vname.reshape(self.n,1)
+
+        return data, vname
+
+
+    # def create_subgroup_labels(self):
+    #     """
+    #     Create unique integer id for each stratum x cluster combination
+        
+    #     The ids are 0, 1, ... and are kept as an attribute
+    #     """
+
+    #     _, i1 = np.unique(self.strata, return_inverse=True)
+    #     _, i2 = np.unique(self.cluster, return_inverse=True)
+    #     labels = i1 * len(i2) + i2
+    #     self.grp_ix = dict.fromkeys(np.unique(labels),[])
+    #     # doesnt append correctly
+    #     for i, k in enumerate(labels):
+    #         self.grp_ix[k].append(i)
+    #     return self.group_ix
+
+    # def get_psu_indices(self):
+    #     self.indices_dict = {}
+    #     unique_strata = np.unique(self.strata)
+    #     stratum_stats = np.empty(len(unique_strata))
+    #     for id, stratum in enumerate(unique_strata):
+    #         # get indices for a particular stratum and the # of clusters in it
+    #         id_stratum = np.where(self.strata == stratum)[0]        
+    #         unique_clusters = np.unique(self.cluster[id_stratum])
+    #         num_clusters = len(unique_clusters)
+    #         self.indices_dict[stratum] = unique_clusters
+    #     # print(self.indices_dict)
+
+    def _jackknife(self, method, column_index, object):
+        unique_strata = np.unique(self.strata)
+        stratum_stats = np.empty(len(unique_strata))
+        for id, stratum in enumerate(unique_strata):
+            # get indices for a particular stratum and the # of clusters in it
+            id_stratum = np.where(self.strata == stratum)[0]        
+            unique_clusters = np.unique(self.cluster[id_stratum])
+            num_clusters = len(unique_clusters)
+
+            # get the statistic without a particular cluster
+            cluster_stats = np.empty(num_clusters)
+
+            for ind, cluster in enumerate(unique_clusters):
+        # self.get_psu_indices()
+        # stratum_stats = np.empty(len(self.indices_dict.keys()))
+        # for id, stratum in enumerate(self.indices_dict.keys()):
+        #     id_stratum = np.where(self.strata == stratum)[0]
+        #     num_clusters = len(self.indices_dict[stratum])
+        #     cluster_stats = np.empty(num_clusters)
+        #     for ind, cluster in enumerate(self.indices_dict[key]):
+                new_weights = self._reweight(cluster, id_stratum, method)
+
+                cluster_stats[ind] = object._stat(new_weights)
+
+            stratum_stats[id] = np.sum((cluster_stats) ** 2)
+            stratum_stats[id] *= (num_clusters - 1) / num_clusters
+        return np.sqrt(np.sum(stratum_stats))
+
+            # diff = (cluster_stats - method) * (num_clusters) / (num_clusters - 1)
+            # stratum_stats[id] = sum(diff)
+        
+
+    def _reweight(self, cluster, id_stratum, method):
```

Comment:
I see. id_stratum hold the indices for the stratum we are 'at', and `intersect1d` is used to find the indices that are either multipled by n_h / (n_h - 1) or turned to 0. `intersect1d` solved the issue of keeping the original indices: Lets say strata is [1 2 1 2 1 2 1 2] so indices where strata == 1 yields [0 2 4 6] and cluster is just a corresponding array. When I do cluster[0 2 4 6], the possible indices to extract from there are [0 1 2 3], instead of the desired [0 2 4 6], so the indices returned wouldn't match up.